cocos/renderer/CCRenderer.h

```diff
@@ -182,6 +184,8 @@ class CC_DLL Renderer
 
     /** set color for clear screen */
     void setClearColor(const Color4F& clearColor);
+    /* returns the Render Queue */
+    std::vector<RenderQueue>& getRenderGroups() { return _renderGroups; }
```

Comment:
I don't like the idea of just handing out a mutable reference of the render groups to the world, I think it's poor encapsulation. Can we move the logic you've added to CCRenderTexture into a helper function in CCRenderer? I think that will be better encapsulated.